src/vcpkg-utils.ts

```diff
@@ -177,7 +169,25 @@ export class Utils {
   }
 
   public static getAllCachedPaths(baselib: baselib.BaseLib, vcpkgRootDir: string): string[] {
-    let paths = runvcpkglib.getOrdinaryCachedPaths(vcpkgRootDir);
+    const runnerOS = process.env.ImageOS ? process.env.ImageOS : process.platform;
+    const isWindows = runnerOS.slice(0,3) == "win";
+
+    let directorySeparator = '/';
+    if (isWindows) {
+      directorySeparator = '\\';
+    }
+
+    let paths = [ vcpkgRootDir,
+                  "!" + vcpkgRootDir + directorySeparator + "installed",
+                  "!" + vcpkgRootDir + directorySeparator + "packages",
+                  "!" + vcpkgRootDir + directorySeparator + "buildtrees",
+                  "!" + vcpkgRootDir + directorySeparator + "downloads" ];
+
+    if (isWindows) {
```

Comment:
I don't think there's a need to add more complexity to this action. vcpkg already allows the user to control the sources used for binary caching with the [VCPKG_BINARY_SOURCES environment variable](https://vcpkg.readthedocs.io/en/latest/specifications/binarycaching/#configuration). I think this action should unconditionally cache the default directories of the binary caching. If the user wants to use a different path for whatever reason, that path can be specified with VCPKG_BINARY_SOURCES and passed to the `additionalCachedPaths` parameter of this action. In this case I think the default location would be empty so there wouldn't be a harm in caching it. If vcpkg does still fall back to the default cache path, the user can add `clear` to the end of their VCPKG_BINARY_SOURCES environment variable.

Reply:
i think my other comment about `VCPKG_DEFAULT_BINARY_CACHE` is valid here as well.  But since GitHub view is not really helpful in highlighting the exact lines this comment was referring too.